doc/policy_tutorial/POLICY_RUN_LOCALLY.md

```diff
@@ -0,0 +1,149 @@
+
+```shell
+$ docker exec -it 5a72c49671c5 /bin/bash
+```
+
+Now we have another interactive bash shell in the APIcast development container and we can issue the HTTP request to test the policy from here.
+
+In the container issue the following HTTP request:
+
+```shell
+$ curl localhost:8080
+<html>
+<head><title>403 Forbidden</title></head>
+<body bgcolor="white">
+<center><h1>403 Forbidden</h1></center>
+<hr><center>openresty/1.13.6.2</center>
+</body>
+</html>
+```
+
+The response will be a 403 Forbidden. Let’s look at the logs to see what has happened.
+
+```shell
+$ bin/apicast --log-level=notice --dev -c examples/configuration/hello_world_config.json
+loading production environment configuration: /home/centos/gateway/config/production.lua
+loading development environment configuration: /home/centos/gateway/config/development.lua
+2019/04/29 09:32:33 [notice] 257#257: [lua] environment.lua:194: add(): loading environment configuration: /home/centos/gateway/config/production.lua
+2019/04/29 09:32:33 [notice] 257#257: [lua] environment.lua:194: add(): loading environment configuration: /home/centos/gateway/config/development.lua
+2019/04/29 09:32:33 [notice] 257#257: using the "epoll" event method
+2019/04/29 09:32:33 [notice] 257#257: openresty/1.13.6.2
+2019/04/29 09:32:33 [notice] 257#257: built by gcc 4.8.5 20150623 (Red Hat 4.8.5-28) (GCC)
+2019/04/29 09:32:33 [notice] 257#257: OS: Linux 4.4.0-146-generic
+2019/04/29 09:32:33 [notice] 257#257: getrlimit(RLIMIT_NOFILE): 1048576:1048576
+2019/04/29 09:43:17 [notice] 257#257: *6 [lua] hello_world.lua:53: request is not authorized, secrets do not match, client: 127.0.0.1, server: _, request: "GET / HTTP/1.1", host: "localhost:8080"
+[29/Apr/2019:09:43:17 +0000] localhost:8080 127.0.0.1:40946 "GET / HTTP/1.1" 403 175 (0.000) 0
+```
+
+The APIcast is running with the notice log-level, when started with the debug log-level considerable more log events will be present. But the one we are interested in is the second from the bottom.
+
+Which states: **‘request is not authorized, secrets do not match’** which is put in the log by the following line in the policy code:
+
+```lua
+if secret_header ~= self.secret then
+ ngx.log(ngx.NOTICE, "request is not authorized, secrets do not match")
+ ngx.status = 403
+ return ngx.exit(ngx.status)
+```
+
+So we now our policy is executing. Let’s provide the secret header in our request in order to pass the validation. Issue the following HTTP request:
+
+```shell
+$ curl localhost:8080 -H 'secret: mysecret'
+GET / HTTP/1.1
+X-Real-IP: 127.0.0.1
+Host: echo
+User-Agent: curl/7.29.0
+Accept: */*
+secret: mysecret
+```
+
+Now we received a valid 200 response from the echo server. But the actual rewrite of query parameters to header is not tested, since the request did not contain any query parameters. So issue a new request with a query parameter to see the transformation at work. Issue the following request:
```

Comment:
```suggestion
You should receive a valid 200 response from the echo server. The rewrite of query parameters to header is not tested, since the request did not contain any query parameters. Issue a new request with a query parameter to see the transformation at work. Issue the following request:
```